### ISSUE TITLE
rephrase warning and error messages

### ISSUE THREAD

**Issue by MichaelPFrey** [openscad/openscad#2780]
some of our warnings can be misleadingly worded, depending on how they are triggered:
- WARNING: Recursion detected calling module
   - this is triggered by `if (StackCheck::inst()->check()) {`, which checks for stack usage, which offcourse can also be triggered why stack exhaustion.
- WARNING: Ignoring unknown variable
   - No, the variable is not ignored, "just" assumed tobe undef
   - The variable also is "not initialized" or "not declared"

**Comment by jordanbrown0** [comment]
> Recursion detected

Indeed, recursion per se is completely normal and expected.  What this is really detecting is **excessive** recursion, as evidenced by hitting a yellow zone on the stack depth.  Perhaps "Excessive recursion" or "Stack overflow".

> ignoring unknown variable

"not initialized", "not declared", and "unknown" mean the same thing to me.  (not-initialized is different in languages with declarations, but OpenSCAD doesn't have declarations.)  But indeed "ignored" is questionable.  Perhaps "unknown variable defaulted to undef".

It would be clever to do spelling correction and offer possible replacements, but that's a more involved project.

**Comment by jordanbrown0** [comment]
In theory we could detect when a variable (or function or module) is present somewhere in the program, but not currently in scope, and give that a different message. However, in practice that would require some rearchitecture because we don't maintain a global symbol table, just a stack of the currently-in-scope names.
